src/main/java/com/google/sps/servlets/FindLocationServlet.java

```diff
@@ -19,7 +27,25 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    //TODO - implement the rest of this function (@Kaleab)
+    String correlator = request.getParameter("correlator");
+
+    //TODO: find a more graceful way to handle this
+    if (correlator == null){
+      throw new IOException();
```

Comment:
you should add a string message to the IOException.

Or by contrast, you could just throw no exception at all.  It looks like what you are trying to do is validate incoming data.

If the incoming data is not in the correct format it could be just spam, or not from your service.  In that case just exiting this function and doing nothing seems like the correct course of action.

Perhaps make another function called validRequest.

if (!validRequest(request)) {
  // The request is not from our service or has invalid data.
  return;
}

Reply:
I return a 404 response code now!